docs/forms.md

```diff
@@ -0,0 +1,143 @@
+# AcroForms in PDFKit
```

Comment:
Maybe we could just refer to them as "Forms" rather than "AcroForms" for approachability?